experimental/GITFans/GITFans.jl

```diff
@@ -0,0 +1,601 @@
+    q_cone_int_point = get_interior_point(q_cone)
+
+    start_hash = compute_bit_list(orbit_list, q_cone_int_point)
+    orbit_start_hash_smallest = find_smallest_orbit_element(
+        start_hash, generators_new_perm, bitlist_oper_tuple, ==,
+        less_or_equal_array_bitlist)
+    hash_list = [orbit_start_hash_smallest]
+
+    current_pos = 0
+    edges = Set(Vector{Int}[])
+
+    for current_hash in hash_list
+        current_pos = current_pos + 1
+
+        # note that we run also over elements added inside the loop
+        current_cone_list = cones_from_bitlist(orbit_list, current_hash)
+        intersected_cone = Polymake.polytope.intersection(current_cone_list...)
+        facets = intersected_cone.FACETS
+        facets = convert(Matrix{Rational{BigInt}},facets)
+        facet_points = Vector{Rational{BigInt}}[
+                           Polymake.polytope.facet(intersected_cone, i-1).REL_INT_POINT
+                           for i in 1:size(facets, 1)]
+
+        neighbor_hashes = []
+        for i in 1:length(facet_points)
+            if any(x->x==0, q_cone_facets_converted*facet_points[i])
+                continue
+            end
+            push!(neighbor_hashes, get_neighbor_hash(orbit_list, facet_points[i], facets[i, :]))
+        end
+
+        neighbor_hashes = map(i->find_smallest_orbit_element(i, generators_new_perm, bitlist_oper_tuple, ==, less_or_equal_array_bitlist), neighbor_hashes)
+        for i in neighbor_hashes
+            if i in hash_list
+                # perhaps we have found a new incidence
+                push!(edges, sort([findfirst(x->x == i, hash_list), current_pos]))
+            else
+                # new representative found
+                push!(hash_list, i)
+                # new incidence
+                push!(edges, [current_pos, length(hash_list)])
+            end
+        end
+    end
+
+    return (hash_list, edges)
+end;
+
+
+orbits_of_maximal_GIT_cones(orbit_list, hash_list, matrix_action) =
+    orbit_cone_orbits( map(x -> hash_to_cone(orbit_list, x), hash_list), matrix_action);
+
+
+function hashes_to_polyhedral_fan(orbit_list, hash_list, hom)
+    # translate the descriptions of the orbit repres. of maximal cones
+    # to cone objects
+    result_cones = map(x -> hash_to_cone(orbit_list, x), hash_list)
+
+    # expand their orbits
+    expanded = orbit_cone_orbits(result_cones, hom)
+    maxcones = vcat(expanded...)
+
+    # the defining rays for all maximal cones
+    rays_maxcones = [[convert(Vector{Rational{BigInt}}, cone.RAYS[i, :])
+                      for i in 1:size(cone.RAYS, 1)]
+                      for cone in maxcones]
+
+    # the defining rays for the orbit representatives of maximal cones
+    rays_result_cones = [[convert(Vector{Rational{BigInt}}, cone.RAYS[i, :])
+                      for i in 1:size(cone.RAYS, 1)]
+                      for cone in result_cones]
+
+    # the set of rays
+    allrays = sort(collect(Set(vcat(rays_maxcones...))))
```

Comment:
What is the `collect(Set(...))` bit about? Is this to get a duplicate free list? Then perhaps `unique` could be used instead? And then `sort! instead of `sort` to avoid yet another allocation?
```suggestion
    allrays = sort!(unique(vcat(rays_maxcones...)))
```